pkg/autodiscovery/autoconfig.go

```diff
@@ -244,6 +244,10 @@ func (ac *AutoConfig) GetAllConfigs() []integration.Config {
 func (ac *AutoConfig) getChecksFromConfigs(configs []integration.Config, populateCache bool) []check.Check {
 	allChecks := []check.Check{}
 	for _, config := range configs {
+		if config.MetricConfig == nil && len(config.Instances) == 0 {
+			// skip non check configs.
```

Comment:
maybe we should have some utility method on `[]integration.Config` to tell if it's a check config

Reply:
I think it should be the responsibility of the check package to validate check configurations, WDYT ?